makefiles/kconfig.mk

```diff
@@ -0,0 +1,90 @@
+# This is the root Kconfig
+KCONFIG ?= $(RIOTBASE)/Kconfig
+
+# Include tools targets
+include $(RIOTMAKE)/tools/kconfiglib.inc.mk
+
+# Generated dir will contain Kconfig generated configurations
+GENERATED_DIR = $(BINDIR)/generated
+
+# This file will contain all generated configuration from kconfig
+export KCONFIG_GENERATED_AUTOCONF_HEADER_C = $(GENERATED_DIR)/autoconf.h
+
+# Add configuration header to build dependencies
+BUILDDEPS += $(KCONFIG_GENERATED_AUTOCONF_HEADER_C)
+
+# Include configuration header when building
+CFLAGS += -include '$(KCONFIG_GENERATED_AUTOCONF_HEADER_C)'
+
+# Header for the generated header file
+define KCONFIG_AUTOHEADER_HEADER
+/* RIOT Configuration File */
+
+endef
+export KCONFIG_AUTOHEADER_HEADER
+
+# This file will contain the calculated dependencies formated in Kconfig
+export KCONFIG_GENERATED_DEPENDENCIES = $(GENERATED_DIR)/Kconfig.dep
+
+# This file will contain application default configurations
+KCONFIG_APP_CONFIG = $(APPDIR)/app.config
+
+# Default and user overwritten configurations
+KCONFIG_USER_CONFIG = $(APPDIR)/user.config
+
+# This file will contain merged configurations from MERGE_SOURCES and is the
+# one that is used to generate the 'riotconf.h' header
+KCONFIG_MERGED_CONFIG = $(GENERATED_DIR)/merged.config
+
+# Flag that indicates that the configuration has been edited
+KCONFIG_EDITED_CONFIG = $(GENERATED_DIR)/.editedconfig
+
+# Add configurations to merge, in ascendent priority (i.e. a file overrides the
+# previous ones)
+MERGE_SOURCES += $(wildcard $(KCONFIG_APP_CONFIG))
+MERGE_SOURCES += $(wildcard $(KCONFIG_USER_CONFIG))
+
+# Create directory to place generated files
+$(GENERATED_DIR):
+	mkdir -p $@ && echo "Created"
```

Comment:
The current problem why murdock fails is that the `generated` directory is deleted. I did read up on [chained rules](https://www.gnu.org/software/make/manual/html_node/Chained-Rules.html#Chained-Rules) in the GNU Make doc. The problem is that `make` considers the `generated` folder as an intermediate file. Intermediate files are usually deleted by `make` after a chain. To tell `make` to keep the folder around, we therefore need to use [.SECONDARY or .PRECIOUS](https://www.gnu.org/software/make/manual/html_node/Special-Targets.html).

So something like
```
.SECONDARY: $(GENERATED_DIR)
$(GENERATED_DIR):
    mkdir -p $@
```
should do the trick ..

Reply:
@leandrolanzieri Let's try to use `.SECONDARY` on the header file itself, first. That may be hint enough for `make` to not delete the folder.

Reply:
Does not seem to solve the issue. I will push again using `.SECONDARY` on the directory.